### PR TITLE
Quad 138

### PR DESCRIPTION
**Pull request by JNagarkar**
Fetching metadata of transformation files uploaded and displaying on transformations page. 
This is a pull request for QUAD-122, QUAD-140, QUAD-138

### CODE REVIEW

File: Quadriga/pom.xml [hunk 1]
```diff
@@ -255,6 +255,16 @@
...
 		</dependency>
 		<dependency>
+			<groupId>commons-io</groupId>
+			<artifactId>commons-io</artifactId>
+			<version>2.4</version>
+		</dependency>		
+		<dependency>
+			<groupId>commons-fileupload</groupId>
+			<artifactId>commons-fileupload</artifactId>
+			<version>1.2.2</version>
```
Comment (jdamerow) [inline Quadriga/pom.xml]: still not the latest version

File: Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/transform/TransformFilesDAO.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		saveNewDTO(tranformDTO);				
+	}
+	
+	@Override
+	public List<TransformFilesDTO> getAllTransformations(){
+		List<TransformFilesDTO> transformList = sessionFactory.getCurrentSession().createCriteria(TransformFilesDTO.class).list();
+		return transformList;
+	}
+
+
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/transform/TransformFilesDAO.java]: code style

File: Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/transform/TransformFilesDAO.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+public class TransformFilesDAO extends BaseDAO implements
+		ITransformFilesDAO {
+
+	@Override
+	public Object getDTO(String id) {		
+		return getDTO(TransformFilesDTO.class, id);
+	}
+
+	@Override
+	public void saveTransformDTO(TransformFilesDTO tranformDTO) {
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/transform/TransformFilesDAO.java]: remove, this class doesn't do anything else than calling an exising method.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/transform/TransformFilesDAO.java (hunk 2 above)
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/transform/TransformFilesDAO.java]: javadoc missing

File: Quadriga/src/main/java/edu/asu/spring/quadriga/dto/TransformFilesDTO.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+@XmlRootElement
+public class TransformFilesDTO implements Serializable {
+
+	private static final long serialVersionUID = 1L;
+	@Id
+	private String Id;
+	@Column(name="Title")
+	String title;
+	@Column(name="Description")
+	String description;
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/dto/TransformFilesDTO.java]: should be private

File: Quadriga/src/main/java/edu/asu/spring/quadriga/dto/TransformFilesDTO.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+/**
+ * DTO for holding the metadata of transformation files
+ * 
+ * @author JayaVenkat
+ *
+ */
+@Entity
+@Table(name = "tbl_transfomationfiles_metadata")
+@XmlRootElement
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/dto/TransformFilesDTO.java]: why is this xmlrootelement here?

File: Quadriga/src/main/java/edu/asu/spring/quadriga/dto/TransformFilesDTO.java [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+	public void setMappingFile(String mappingFileName) {
+		this.mappingFileName = mappingFileName;
+	}
+
+	public static long getSerialversionuid() {
+		return serialVersionUID;
+	}
+
+
+}
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/dto/TransformFilesDTO.java:135]: code style

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/transformation/TransformationManager.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+	}
+	
+	@Transactional
+	@Override
+	public List<TransformFilesDTO> getTransformationsList(){
+		return transformationDAO.getAllTransformations();
+	}
+
+}
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/transformation/TransformationManager.java:43]: code style

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/transformation/ShowTranformationFilesUiController.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+@Controller
+public class ShowTranformationFilesUiController {
+	
+	/**
+	 * 
+	 * This method will show uploadTranformation.jsp to user 
+	 * @return
+	 */
+	@RequestMapping( value="auth/transformation/selectTransformationFiles",method=RequestMethod.GET)	
+	public String selectTransformationFiles(HttpServletRequest request){
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/transformation/ShowTranformationFilesUiController.java]: why do you inject a request object here? 

Also, is this the method that shows the list of transformations? 

path should have project id in it like the other pages.
Project owner, project admins, and collaborators should have access to this, so you need to use an access policy.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/transformation/TransformFilesBackingBean.java [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+	}
+
+	public String getPatternDescription() {
+		return patternDescription;
+	}
+
+	public void setPatternDescription(String patternDescription) {
+		this.patternDescription = patternDescription;
+	}
+
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/transformation/TransformFilesBackingBean.java:65]: code style

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/transformation/TransformationAssignUserController.java [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+ * @author: Jaydatta Nagarkar.
+ * 
+ */
+@Controller
+public class TransformationAssignUserController {
+	@Autowired
+    private ITransformationManager transformManager; 
+	    
+	@Autowired
+    private INetworkManager networkManager;
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/transformation/TransformationAssignUserController.java:46]: what's up with the indentations?

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/transformation/TransformationAssignUserController.java [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+     * @param model
+     * @param principal
+     * @return
+     * @throws QuadrigaStorageException
+     */
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT, paramIndex = 1, userRole = {
+            RoleNames.ROLE_PROJ_COLLABORATOR_CONTRIBUTOR,
+            RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN,
+            RoleNames.ROLE_QUADRIGA_ADMIN}) })
+    @RequestMapping(value = "auth/transformation", method = RequestMethod.GET)
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/transformation/TransformationAssignUserController.java:72]: path should contain project id

Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/transformation/TransformationAssignUserController.java:72]: transformations are depending on project not user

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/transformation/UploadTransformFilesController.java [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+ * This is a controller which takes request from uploadTransfomation.jsp to upload transformation files
+ */
+@Controller
+public class UploadTransformFilesController {
+
+	@Autowired
+	private ITransformationManager transformationManager;
+	
+	@RequestMapping(value="auth/transformation/upload",method=RequestMethod.POST)
+	public String uploadTransformFiles(@ModelAttribute("TransformationFilesBackingBean") TransformFilesBackingBean formBean, ModelMap map,
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/transformation/UploadTransformFilesController.java]: formBean should be validated and make sure files are not null

**Comment by jdamerow** [comment]
please read the style guide and follow it

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/transformation/ShowTranformationFilesUiController.java (hunk 8 above)
Comment (jvutukur) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/transformation/ShowTranformationFilesUiController.java]: This method will show jsp page from which user can upload transformation files. 
This method **is not for showing** a page which contains  list of transfomations

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/transformation/TransformationAssignUserController.java (hunk 11 above)
Comment (JNagarkar) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/transformation/TransformationAssignUserController.java:72]: This method just populates the existing approved projects, and displays on transformation page.  Name changed to TransformProjectsController

**Pull request closed without merging**